jetty-server/src/test/java/org/eclipse/jetty/server/ErrorHandlerTest.java

```diff
@@ -72,8 +79,33 @@ protected void generateAcceptableResponse(
                     default:
                         super.generateAcceptableResponse(baseRequest,request,response,code,message,mimeType);
                 }
+            }    
+        });
+        
+        server.setHandler(new AbstractHandler()
+        {
+            @Override
+            public void handle(String target, Request baseRequest, HttpServletRequest request, HttpServletResponse response) throws IOException, ServletException
+            {
+                if (baseRequest.getDispatcherType()==DispatcherType.ERROR)
+                {
+                    baseRequest.setHandled(true);
+                    response.sendError(((Integer)request.getAttribute(RequestDispatcher.ERROR_STATUS_CODE)).intValue());
+                    return;
+                }
+                
+                if(target.startsWith("/charencoding/"))
+                {
+                    baseRequest.setHandled(true);
+                    response.setCharacterEncoding("utf-8");
+                    response.sendError(404);
```

Comment:
I would prefer a `return` statement here, to avoid fall through. It's a classic that we modify the test case, forget about the fall through, and the test does things we don't expect.